src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -301,6 +306,34 @@ def is_saveable_module(name, value):
 
             save_method(os.path.join(save_directory, pipeline_component_name), **save_kwargs)
 
+            if dduf_format:
+                import shutil
+                import tarfile
+
+                dduf_file_path = os.path.join(save_directory, dduf_filename)
+
+                if os.path.isdir(dduf_file_path):
+                    logger.warning(
+                        f"Removing the existing folder {dduf_file_path} so that we can save the DDUF archive."
+                    )
+                    shutil.rmtree(dduf_file_path)
+                if (
+                    os.path.exists(dduf_file_path)
+                    and os.path.isfile(dduf_file_path)
+                    and tarfile.is_tarfile(dduf_file_path)
+                ):
+                    # Open in append mode if the file exists
+                    mode = "a"
+                else:
+                    # Open in write mode to create it if it doesn't exist
+                    mode = "w:"
+                with tarfile.open(dduf_file_path, mode) as tar:
+                    dir_to_archive = os.path.join(save_directory, pipeline_component_name)
+                    if os.path.isdir(dir_to_archive):
+                        tar.add(dir_to_archive, arcname=os.path.basename(dir_to_archive))
+                        # remove from save_directory after we added it to the archive
+                        shutil.rmtree(dir_to_archive)
```

Comment:
Might wanna consider using a ZIP archive instead, as also mentioned over Slack.

Reply:
Also, expanding the archive and removing it could be problematic.

- If I understand correctly, we would be removing the symlink to the blob, but not the blob itself. We won't be saving disk space.
- If we did remove the blob, I think it'd trigger another download next time we call `.from_pretrained`.
- Removing the reference could maybe have consequences when checking for newer versions in the Hub (not sure).

When we change to using zips, I wonder if we could extract files progressively in memory during the load process, and keeping the big .dduf file as is. Having a disk snapshot is convenient, but I can see no easy solution to avoid duplicating space.

Reply:
Switched to zip !